packages/fxa-content-server/app/scripts/views/settings.js

```diff
@@ -142,7 +150,17 @@ const View = BaseView.extend({
   beforeRender() {
     const account = this.getSignedInAccount();
 
-    return account.fetchProfile().then(() => this.user.setAccount(account));
+    return account
+      .fetchProfile()
+      .then(() => this.user.setAccount(account))
+      .then(() => account.settingsData())
```

Comment:
Calling `fetchProfile` and `settingsData` serially will delay render, especially on slower devices. Can you do a `Promise.all` and make both requests at once?

Reply:
Yep, and actually, I just noticed that `user.setAccount` calls `account.fetch`, which calls `account.sessionStatus`, which is also a remote request. So I'll `Promise.all` all three...